dap_aria_mapping/analysis/app/pages/1_Horizon_Scanner.py

```diff
@@ -0,0 +1,307 @@
+    unique_topics = list(list(volume_data.select(pl.col("topic_name").unique()))[0])
+    return volume_data, alignment_data, unique_topics
+
+def group_emergence_by_level(_volume_data: pl.DataFrame, level: str, y_col: str) -> pl.DataFrame:
+    """groups the data for the emergence chart by the level specified by the filters
+
+    Args:
+        _volume_data (pl.DataFrame): data for backend of emergence chart
+        level (str): level to view, specified by domain/area filters
+        y_col (str): patents, publications, or all documents (specified by filter)
+
+    Returns:
+        pl.DataFrame: grouped emergence data for chart
+    """
+    q = (_volume_data.lazy().with_columns(
+        pl.col(level).cast(str)
+        ).groupby(
+            [level, "{}_name".format(level),"year"]
+            ).agg(
+                [pl.sum(y_col)]
+                ).filter(pl.any(pl.col("year").is_not_null())))
+    return q.collect()
+
+def group_alignment_by_level(_alignment_data: pl.DataFrame, level: str) -> pl.DataFrame:
+    """groups the data for the alignment chart by the level specified by the filters.
+    Also calculates the fraction of total documents per type to visualise in the chart.
+
+    Args:
+        _alignment_data (pl.DataFrame): data for backend of alignment chart
+        level (str): level to view, specified by domain/area filters
+
+    Returns:
+        pl.DataFrame: grouped alignment data for chart
+    """
+    total_pubs = _alignment_data.filter(pl.col("doc_type")=="publication_count").select(pl.sum("count"))
+    total_patents = _alignment_data.filter(pl.col("doc_type")=="patent_count").select(pl.sum("count"))
+    q = (_alignment_data.lazy().with_columns(
+        pl.col(level).cast(str)
+        ).groupby(["doc_type", level, "{}_name".format(level)]
+        ).agg(
+            [pl.sum("count").alias("total")]
+        ).with_columns(
+            pl.when(pl.col("doc_type") == "publication_count")
+            .then(pl.col("total")/total_pubs)
+            .when(pl.col("doc_type") == "patent_count")
+            .then(pl.col("total")/total_patents)
+            .alias("doc_fraction")
+        ).with_columns(
+            pl.when(pl.col("doc_type") == "publication_count")
+            .then("Publications")
+            .when(pl.col("doc_type") == "patent_count")
+            .then("Patents")
+            .alias("doc_name_clean")
+        )
+        .with_columns(
+            (pl.col("doc_fraction")*100).alias("doc_percentage"))
+        )
+    return q.collect()
+
+def convert_to_pandas(_df: pl.DataFrame) -> pd.DataFrame:
+    """converts polars dataframe to pandas dataframe
+    note: this is needed as altair doesn't allow polars, but the conversion is quick so i still think it's 
+    worth while to use polars for the filtering
+
+    Args:
+        _df (pl.DataFrame): polars dataframe
+
+    Returns:
+        pd.DataFrame: pandas dataframe
+    """
+    return _df.to_pandas()
```

Comment:
Perhaps this should go into `utils/app_data_utils.py` ?

Reply:
Yep good suggestion - `app_data_utils` was meant to support the pipelines, so I created a separate `app_utils` for frontent utils and specified the difference in the docstring